pkg/managed/services_test.go

```diff
@@ -0,0 +1,99 @@
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create service: %v", err))
+	}
+
+	for clusterName := range testutil.GetModelBindingPair().ManagedClusters {
+		managedClusterConnection := testutil.GetManagedClusterConnections()[clusterName]
+		existingServices, err := managedClusterConnection.ServiceLister.Services(constants.MonitoringNamespace).List(labels.Everything())
+		if err != nil {
+			t.Fatal(fmt.Sprintf("can't list service for cluster %s: %v", err, clusterName))
+		}
+		assert.Equal(t, 0, len(existingServices), "no services should be found for cluster %s", clusterName)
+	}
+}
+
+// TestCreateServicesVMIBinding model/binding for VMI system binding - Node Exporter service is created.
+func TestCreateServicesVMIBinding(t *testing.T) {
+	mbPairSystemBinding := testutil.GetModelBindingPair()
+	mbPairSystemBinding.Model.Name = constants.VmiSystemBindingName
+	mbPairSystemBinding.Binding.Name = constants.VmiSystemBindingName
+
+	// Model/binding for VMI system binding - Node Exporter service is created.
+	managedConnections := testutil.GetManagedClusterConnections()
+	err := CreateServices(mbPairSystemBinding, managedConnections)
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create service: %v", err))
+	}
+
+	for clusterName := range mbPairSystemBinding.ManagedClusters {
+		managedClusterConnection := managedConnections[clusterName]
+		existingService, err := managedClusterConnection.KubeClient.CoreV1().Services(constants.MonitoringNamespace).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			t.Fatal(fmt.Sprintf("can't list service for cluster %s: %v", err, clusterName))
+		}
+		assertCreateService(t, existingService, clusterName)
+
+		// Update the port value so when we call CreateServices again the update code is executed.
+		existingService.Items[0].Spec.Ports[0].Port = 9200
+		_, err = managedClusterConnection.KubeClient.CoreV1().Services(constants.MonitoringNamespace).Update(context.TODO(), &existingService.Items[0], metav1.UpdateOptions{})
+		if err != nil {
+			t.Fatal(fmt.Sprintf("can't update service for cluster %s: %v", err, clusterName))
+		}
+	}
+
+	// Model/binding for VMI system binding - service already exist
+	err = CreateServices(mbPairSystemBinding, managedConnections)
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create service: %v", err))
+	}
+
+	for clusterName := range mbPairSystemBinding.ManagedClusters {
+		managedClusterConnection := managedConnections[clusterName]
+		existingService, err := managedClusterConnection.KubeClient.CoreV1().Services(constants.MonitoringNamespace).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			t.Fatal(fmt.Sprintf("can't list service for cluster %s: %v", err, clusterName))
+		}
+		assertCreateService(t, existingService, clusterName)
+	}
+}
+
+func assertCreateService(t *testing.T, existingService *corev1.ServiceList, clusterName string) {
+	labels := map[string]string{
+		constants.ServiceAppLabel: constants.NodeExporterName,
+	}
+
```

Comment:
You can do this to avoid passing t to every assert
	assert := assert.New(t)
        assert.Equal(1,...